### PR TITLE
Ubuntu 14.04

### PR DESCRIPTION
**Pull request by sreedishps**
Changes for making the build for ubuntu 14.04

### CODE REVIEW

File: README-UBUNTU-14.04 [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Setup steps - non hdfs
+1)cd scribe; rm -f debian; ln -sf debian_nonhdfs debian
+2)sudo vi /etc/apt/sources.list.d/inmobi.list
+add the entry deb http://repo.corp.inmobi.com:9999/precise ./
+3)sudo apt-get update
+4)sudo apt-get install debhelper autotools-dev inmobi-techplatform-libthrift-dev inmobi-techplatform-libfb303 libboost-thread-dev  libstdc++6-4.4-dev libevent-dev inmobi-techplatform-thrift-compiler libboost-dev libboost-test-dev libboost-program-options-dev libboost-system-dev libboost-filesystem-dev libevent-dev automake libtool flex bison pkg-config g++ libssl-dev
+5) creating sym links for pointing the multi threaded libboost to single threaded libboost
```
Comment (satish-mittal) [inline README-UBUNTU-14.04:7]: Why is this changed needed?

File: configure.ac [hunk 2]
```diff
@@ -10,7 +10,7 @@
 AC_PREREQ(2.52)
 AC_INIT([scribe], [1.5.0])
 AC_CONFIG_MACRO_DIR([aclocal])
-AM_INIT_AUTOMAKE([foreign -Wall])
+#AM_INIT_AUTOMAKE([foreign -Wall])
```
Comment (satish-mittal) [inline configure.ac]: 1) Why do we need to remove foreign -Wall option?

File: README-UBUNTU-14.04 (hunk 1 above)
Comment (sreedishps) [inline README-UBUNTU-14.04:7]: scribe expects multi threaded libboost and is currently deprecated. So need to create symlinks as explained in this post
http://askubuntu.com/questions/486006/cannot-find-boost-thread-mt-library

File: configure.ac (hunk 2 above)
Comment (sreedishps) [inline configure.ac]: I faced the same issue, and also did the same fix as done in this pull request https://github.com/facebookarchive/scribe/pull/83

From the conversation "The bug is due to Scribe calling AM_INIT_AUTOMAKE in both the top of configure.ac and inside FB_INITIALIZE defined in acinclude.m4. The solution is to comment the AM_INIT_AUTOMAKE"

Comment (satish-mittal) [inline configure.ac]: Then let us remove the commented line.

File: README-UBUNTU-14.04 (hunk 1 above)
Comment (satish-mittal) [inline README-UBUNTU-14.04:7]: I tried to do some quick search and could find boost multi-threaded libraries for ubuntu 14:

http://packages.ubuntu.com/trusty/libboost-thread1.54-dev
http://packages.ubuntu.com/trusty/libboost-thread1.55-dev
http://stackoverflow.com/questions/26716202/linux-ubuntu-14-04-boost-libs-and-thread-compilations

Scribe is essentially a multi-threaded application. We need to be extremely careful not to impact this aspect.
Also this linking is done for archive files, whereas scribed essentially links with shared libraries.

Comment (sreedishps) [inline README-UBUNTU-14.04:7]: https://groups.google.com/forum/#!topic/boost-list/5nbpE0J_E1c - dicussion on the same. 
http://permalink.gmane.org/gmane.comp.lib.boost.user/62273
libboost-thread-mt and libboost-thread were essentially same and the "-mt" part was a decorator. And now there is no more "-mt" version available in the latest versions of libboost. http://www.boost.org/users/history/version_1_40_0.html
From the build package session. 

"The default naming of libraries in Unix-like environment now matches system conventions, and does not include various decorations"

**Comment by satish-mittal** [comment]
+1

**Comment by satish-mittal** [comment]
@sreedishps 
Pending items:

1) Make sure the same changes are working with scribe collector package as well, or whether we need to fix anything else for collector on ubuntu 14.
2) Finally, edit the changelog for both packages and bump up the major version.
